test/jdk/javax/swing/JSplitPane/TestSplitPaneResetDividerLoc.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8356594
+ * @summary Verifies if JSplitPane loses divider location when
+ *          reopened via JOptionPane.createDialog()
+ * @run main TestSplitPaneResetDividerLoc
+ */
+
+import java.awt.BorderLayout;
+import java.awt.Color;
+import java.awt.Dimension;
+import java.awt.Point;
+import java.awt.Robot;
+import java.awt.event.ActionEvent;
+import java.awt.event.InputEvent;
+import java.awt.event.KeyEvent;
+
+import javax.swing.AbstractAction;
+import javax.swing.JButton;
+import javax.swing.JDialog;
+import javax.swing.JFrame;
+import javax.swing.JOptionPane;
+import javax.swing.JPanel;
+import javax.swing.JSplitPane;
+import javax.swing.SwingUtilities;
+
+public class TestSplitPaneResetDividerLoc {
+
+    private static JPanel lazyPanel;
+    private static JSplitPane splitPane;
+    private static JButton openDialogButton;
+    private static JDialog dialog;
+    private static JFrame frame;
+    private static volatile Point point;
+    private static volatile int setLoc;
+    private static volatile int curLoc;
+
+    public static void main(String[] args) throws Exception {
+        try {
+            SwingUtilities.invokeAndWait(TestSplitPaneResetDividerLoc::createAndShowUI);
+
+            Robot robot = new Robot();
+            robot.waitForIdle();
+            robot.delay(1000);
+            SwingUtilities.invokeAndWait(() -> {
+                point = openDialogButton.getLocationOnScreen();
+            });
+            robot.mouseMove(point.x, point.y);
```

Comment:
When I run the test on my macOS laptop, it fails because the click occured on the window border instead of the button. It would be good to slightly shift the click position to ensure it reliably hits the button across platforms.

btw it is better to use EventQueue.invokeAndWait instead of SwingUtilities.invokeAndWait. The latter was added early on, while the former consolidates all event-dispatching utilities in one class.

Reply:
maybe we can loop over installed L&F?

Reply:
click position fixed......L&F iterated.....